### ISSUE TITLE
Add an Option of Showing Flags in Locale dropdown

### ISSUE THREAD

**Issue by massoudmaboudi** [facebook/docusaurus#7566]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Description

![image](https://user-images.githubusercontent.com/18216265/172035062-fc9505c8-b648-432d-92d0-a4b0cb46e017.png)

It can be an option in docusaurus.config.js to show the flags of each locale as emoji characters for better compatibility.

I have a dilemma for this feature though

- Arguably, the flag of England is the most appropriate flag to represent the English language. But how recognizable is the English flag? 
- However, often the British flag is used to represent English. Within Britain, other languages other than English are spoken — including Welsh and Scottish Gaelic.
- Often the United States flag is used to represent English — and while the USA has far more English speakers than Britain, English does not originate from the United States.

But it may help #7409 also.

### Motivation

To make an improvement in locale dropdown and #7409

### Self-service

- [ ] I'd be willing to contribute this feature to Docusaurus myself.

**Comment by Josh-Cena** [comment]
That looks nice, I think we definitely want this type of stuff.

We might even want to port it to other things, like versions dropdown, navbar link items, sidebar links...

(Note, for navbar link items, we have long thought about using icons on medium-sized screens so people don't have to toggle the sidebar)

**Comment by massoudmaboudi** [comment]
You are right. For example, the ladder icon on docusaurus website is nice. So maybe we can add an option to any dropdown as n optional config.

**Comment by Josh-Cena** [comment]
Ah, that's a construction barrier. And yes, if we make this an API it should be more easily portable.

**Comment by massoudmaboudi** [comment]
It should be able to be placed, on the left, right of the item, and also on the other side of the item like below:
![image](https://user-images.githubusercontent.com/18216265/172035332-1f529aac-4317-4dd5-87a2-9388860ed063.png)
![image](https://user-images.githubusercontent.com/18216265/172035352-f549bdf5-1000-435c-8ee4-6876c94e5f54.png)
![image](https://user-images.githubusercontent.com/18216265/172035410-a8977ab7-8bcb-4e9d-9bde-354647d6e4a0.png)

**Comment by Josh-Cena** [comment]
Mmm, that's a bit complicated. Note also that the external link icon is already taking the right edge, so we'll probably put it on the left edge by default. Not sure if we want a config option to change that—should be swizzlable.

**Comment by massoudmaboudi** [comment]
Or also a bit further, what do you think of the Material Design to be  optional too and if its false, it can show the flag to solve the #7409?

**Comment by massoudmaboudi** [comment]
yes, I raise the ideas, you try to choose the best one.

**Comment by Josh-Cena** [comment]
> the Material Design to be optional too and if its false,

Not sure what you mean exactly, but Material is absolutely on my radar, and why I mentioned " it should probably support strings, HTMLs, and images". We won't officially couple the API to any icon library, but you should be able to inject most kinds of icons.

**Comment by massoudmaboudi** [comment]
![image](https://user-images.githubusercontent.com/18216265/172035589-bfaf5291-f046-4cfc-90ee-fb71e5767156.png)
This Icon, 
Just consider we make it optional if the flag is available. Then it will be like the below:

![image](https://user-images.githubusercontent.com/18216265/172035612-9fab113a-c14a-417b-8d23-840bfc2f437a.png)

**Comment by Josh-Cena** [comment]
Ah, I get you. Not a bad idea? We can think about it once we see the UI/UX.

**Comment by massoudmaboudi** [comment]
Glad to hear that. Waiting for more updates on this.

**Comment by slorber** [comment]
agree we should have this and make a poc

This post provides a very lightweight way to get an emoji from a country code:

https://dev.to/jorik/country-code-to-flag-emoji-a21
